lib/measured/measurable.rb

```diff
@@ -46,13 +48,17 @@ def inspect
   end
 
   def <=>(other)
-    if other.is_a?(self.class) && unit == other.unit
-      value <=> other.value
+    if other.is_a?(self.class)
+      other_converted = other.convert_to(unit)
```

Comment:
Line 26 above allows you to `convert_to` the unit you're already in and it will just return self. So we are converting the `other` to be the current instance's unit and then can do `value` comparistons as scalars, knowing they are all in the same unit.